packages/react-chatgpt-apps/src/hooks/useCallTool.ts

```
@@ -0,0 +1,28 @@
+import { useCallback } from "react";
+import { type CallToolResponse } from "../types.js";
+
+/**
+ * Hook for calling external tools/APIs from within a ChatGPT app.
+ *
+ * This hook provides access to the OpenAI `callTool` function, which allows your
+ * app to invoke external APIs or tools that have been configured in your ChatGPT app.
+ *
+ * @returns A callback function that takes a tool name and arguments, returning the tool's response or null if unavailable
+ *
+ * @example
+ * ```tsx
+ * const callTool = useCallTool();
+ * const response = await callTool("weatherApi", { city: "San Francisco" });
+ * console.log(response?.result);
+ * ```
+ */
+export function useCallTool() {
+  const callTool = useCallback(async (name: string, args: Record<string, unknown>): Promise<CallToolResponse | null> => {
+    if (typeof window !== "undefined" && window?.openai?.callTool) {
```

I think we should be less defensive than this and error loudly instead of failing silently if it isn't defined, can clean up the return type to not have null too

---

Updated to throw an error when the callback is unable to run `callTool`